File: src/main/kotlin/com/taskmanagement/TaskManagement/model/Task.kt
package com.taskmanagement.TaskManagement.model

import org.bson.types.ObjectId
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.time.LocalDateTime

@Document(collection = "tasks")
data class Task(
    @Id
    val id : ObjectId? = null,
    val title : String,
    val description : String,
    val isCompleted : Boolean = false,
    val createdAt : LocalDateTime = LocalDateTime.now()
)

File: frontend/lib/domain/task.dart
class Task {
  final String? id;
  final String title;
  final String description;
  final bool? isCompleted;
  final String? createdAt;
  final String updatedAt;

  Task({
    this.id,
    required this.title,
    required this.description,
    this.isCompleted = false,
    this.createdAt,
    required this.updatedAt,
  });

  factory Task.fromJson(Map<String, dynamic> json) {
    return Task(
      id: json['id'],
      title: json['title'],
      description: json['description'],
      isCompleted: json['isCompleted'] ?? false,
      createdAt: json['createdAt'],
      updatedAt: json['updatedAt'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      if (id != null) 'id': id,
      'title': title,
      'description': description,
      'isCompleted': isCompleted,
      'createdAt': createdAt,
      'updatedAt': updatedAt,
    };
  }
}
